src/tsickle.ts

```diff
@@ -1120,7 +1120,12 @@ class Annotator extends ClosureRewriter {
               (exp.sym.flags & ts.SymbolFlags.Value) === 0;
       if (!isTypeAlias) continue;
```

Comment:
This is actually on purpose, we might need to think a bit about how the right solution should look.

In TypeScript, types are not backed by values. That is, there is no runtime value for the `Foo` in `export type Foo = number;`. When we translate to Closure, some types become values (e.g. interfaces), but some do not (e.g. typedefs).

This is orthogonal to "Ambient". Ambient is something that in the current compilation unit is declared, but not defined - but if there's a `declare class Foo {}`, then there must still be an actual runtime symbol for `Foo` around.

The previous code was attempting to export both type aliases (typedefs in Closure) *and* interfaces using the same syntax, by exporting a typedef that's an alias to the underlying type. In the code below, `Bar` could be either an interface or a typedef:

```
/** @typedef {tsickle_forward_declare_1.Bar} */	+/** @const */
exports.Bar; // re-export typedef
```

Closure (AFAIR) does not support assigning values into `typedef`s, they are supposed to go undefined. So when you're accessing `tsickle_forward_declare_1.Bar` to assign into `Bar`, if there's no value backing it, we might be accessing an undefined value, but worse, create a symbol that's of unknown type to Closure.

Do I understand you correctly that the exported interface does not work for you? In the example that you give, the code contains:

```
// ...
exports.Foo; // re-export typedef
```

Is the problem maybe rather that the `/** @typedef {!tsickle_forward_declare_1.Foo} */` comment that's supposed to go before it is swallowed for some reason?

Reply:
Hi Martin,

Thanks for the reply. Yes if the `typedef` was preserved, I'd be happy, too. It wasn't easy to figure out why that `typedef` is swallowed, though, so just exporting the value seemed like an easier solution.

Any idea on why the `typedef` is swallowed? I'd be happy to submit that fix if you could point me in the right direction.